### ISSUE TITLE
[Tabs]: Disabled tabs are still clickable and active

### ISSUE THREAD

**Issue by vpicone** [carbon-design-system/carbon#2880]
## What package(s) are you using?

- [ ] `carbon-components`
- [x] `carbon-components-react`

## Detailed description

> Describe in detail the issue you're having.

Passing the disabled prop to a tab only impacts the style, the Tabs component will still switch contents when a disabled tab is clicked

> What did you expect to happen? What happened instead? What would you like to
> see changed?

Disabled tabs should have no effect when clicked

## Steps to reproduce the issue

1. Go to [tabs storybook](http://react.carbondesignsystem.com/?selectedKind=Tabs&selectedStory=Default&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Fstories%2Fstories-panel)
2. Under the `knobs` panel of storybook enable the `Disabled` knob
3. Click the tabs, they are still active and change the content of the Tabs component

**Comment by tomkdgun** [comment]
This issue is also important for SPSS Statistics, since we have similar problem.

**Comment by abbeyhrt** [comment]
Hey! I'm gonna try and tackle this one.

**Comment by asudoh** [comment]
Cool to see you are jumping in @abbeyhrt! One suggestion is searching for `pointer-events` in our codebase and see if there is something similar to this case.

**Comment by anibapat** [comment]
@abbeyhrt I was using keyboard nav, arrow keys on Tab component and it makes it active. Could you verify it?

**Comment by abbeyhrt** [comment]
https://codesandbox.io/s/codesandbox-vq8b6 reopening since the content is still switching to disabled tab content when using arrow keys. You also can't navigate passed the disabled tab to a third not disabled tab.
